megamek/src/megamek/common/weapons/ArtilleryWeaponIndirectFireHandler.java

```diff
@@ -455,6 +415,143 @@ else if ((null != bestSpotter) && !(this instanceof ArtilleryWeaponDirectFireHan
         return false;
     }
     
+    /**
+     * Worker function that handles "artillery round landed here" reports,
+     * and direct artillery scatter. 
+     * @return Whether or not we should continue attack resolution afterwards
+     */
+    private Coords handleReportsAndDirectScatter(boolean isFlak, Coords targetPos, Vector<Report> vPhaseReport, ArtilleryAttackAction aaa) {
+        Coords originalTargetPos = targetPos;
+        
+        Report r;
+        // special report for off-board target
+        if (target.isOffBoard()) {
+            r = new Report(9994);
+            r.subject = subjectId;
+            vPhaseReport.addElement(r);
+        }
+        
+        if (!bMissed) {
+            // off-board targets can just report direct hit and move on
+            if (target.isOffBoard()) {
+                r = new Report(9996);
+                r.subject = subjectId;
+                r.indent();
+                vPhaseReport.addElement(r);
+                return targetPos;
+            } 
+            
+            if (!isFlak) {
+                r = new Report(3190);
+            } else {
+                r = new Report(3191);
+            }
+            r.subject = subjectId;
+            r.add(targetPos.getBoardNum());
+            vPhaseReport.addElement(r);
+
+            String artyMsg = "Artillery hit here on round " + game.getRoundCount() 
+                    + ", fired by " + game.getPlayer(aaa.getPlayerId()).getName()
+                    + " (this hex is now an auto-hit)";
+            game.getBoard().addSpecialHexDisplay(
+                    targetPos,
+                    new SpecialHexDisplay(SpecialHexDisplay.Type.ARTILLERY_HIT,
+                            game.getRoundCount(), game.getPlayer(aaa
+                                    .getPlayerId()), artyMsg));
+
+        } else {
+            // direct fire artillery only scatters by one d6
+            // we do this here to avoid duplicating handle()
+            // in the ArtilleryWeaponDirectFireHandler
+            Coords origPos = targetPos;
+            int moF = toHit.getMoS();
+            if (ae.hasAbility("oblique_artillery")) {
+                // getMoS returns a negative MoF
+                // simple math is better so lets make it positive
+                if ((-moF -2) < 1) {
+                    moF = 0;
+                } else {
+                    moF = moF +2;
```

Comment:
Is this just `mof = Math.max(mof + 2, 0)` ?

Reply:
I honestly have no idea, I just copy and pasted it from where it originally was.